crates/query-engine/translation/src/translation/helpers.rs

```diff
@@ -47,16 +47,68 @@ pub struct NativeQueryInfo {
     pub alias: sql::ast::TableAlias,
 }
 
-/// For the root table in the query, and for the current table we are processing,
+/// For the current table we are processing, and all ancestor table up to the closest Query,
 /// We'd like to track what is their reference in the query (the name we can use to address them,
 /// an alias we generate), and what is their name in the metadata (so we can get
 /// their information such as which columns are available for that table).
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct RootAndCurrentTables {
-    /// The root (top-most) table in the query.
-    pub root_table: TableSourceAndReference,
+pub struct TableScope {
```

Comment:
Does this `TableScope` refactor still make sense in `ndc-models` 0.1.0? If so I would really like us to apply it in a separate PR before this one, as it's the source of a lot of changes I can see.

Reply:
If we could split out the `TableScope` changes from the `ndc-models` changes then I think this will become a much clearer atomic change.

Reply:
TableScope is for v0.2, as it enables supporting named scopes. We can apply it in a subsequent PR, but some tests won't pass until then